service/src/main/java/uk/nhs/adaptors/gp2gp/ehr/mapper/EncounterComponentsMapper.java

```diff
@@ -94,99 +93,90 @@ public class EncounterComponentsMapper {
         ResourceType.DiagnosticReport, this::mapDiagnosticReport);
 
     public String mapComponents(Encounter encounter) {
-        Optional<ListResource> listReferencedToEncounter =
-            messageContext.getInputBundleHolder().getListReferencedToEncounter(encounter.getIdElement(), CONSULTATION_LIST_CODE);
-
-        if (listReferencedToEncounter.isEmpty()) {
-            return StringUtils.EMPTY;
-        }
-
-        var entries = listReferencedToEncounter.orElseThrow().getEntry();
-        List<ListResource> topics = entries.stream()
-            .map(entry -> entry.getItem().getReferenceElement())
-            .map(reference ->
-                (ListResource) messageContext
-                    .getInputBundleHolder()
-                    .getRequiredResource(reference))
-            .collect(Collectors.toList());
+        return messageContext
+            .getInputBundleHolder()
+            .getListReferencedToEncounter(encounter.getIdElement(), CONSULTATION_LIST_CODE)
+            .map(this::mapListResourceToComponent)
+            .orElse(StringUtils.EMPTY);
+    }
 
-        return topics.stream()
+    private String mapListResourceToComponent(ListResource listResource) {
```

Comment:
This looks like `mapConsultationListResourceToComponent` (if I'm understanding this) as a more specific name? List is a bit vague.